services/monitor/monitor.sh

```diff
@@ -0,0 +1,156 @@
+
+    cd /workspace
+
+    info "Recreating container with latest image"
+    docker compose -f compose.yaml -f compose.prod.yaml up -d --force-recreate --no-deps sneezy
+
+    sleep 3
+
+    if is_container_running; then
+        success "Container recreated and started successfully"
+        return 0
+    else
+        error "Failed to start container"
+        return 1
+    fi
+}
+
+handle_container_restart() {
+    info "Container stopped - checking for updates"
+
+    if check_for_image_update; then
+        info "Update available - recreating container with new image"
+        send_discord_notification "🔄 SneezyMUD updating to latest version..."
+        if recreate_container; then
+            success "Container updated and restarted successfully"
+            send_discord_notification "✅ SneezyMUD updated to latest version successfully!"
+        else
+            error "Failed to recreate container - attempting simple restart"
+            send_discord_notification "❌ SneezyMUD update failed - falling back to previous version"
+            docker compose -f compose.yaml -f compose.prod.yaml up -d sneezy
+        fi
+    else
+        info "No updates available - restarting with current image"
+        docker compose -f compose.yaml -f compose.prod.yaml up -d sneezy
+    fi
+}
+
+ensure_container_running() {
+    if ! is_container_running; then
+        if container_exists; then
+            local status=$(get_container_status)
+            info "Container exists but not running (status: $status)"
+            handle_container_restart
+        else
+            info "Container does not exist - starting initial container"
+            docker compose -f compose.yaml -f compose.prod.yaml up -d sneezy
+        fi
+    fi
+}
+
+main() {
+    info "Starting SneezyMUD container monitor"
+    info "Monitoring container: $CONTAINER_NAME"
+    info "Check interval: ${CHECK_INTERVAL}s"
+
+    if [ -n "$DISCORD_WEBHOOK_URL" ]; then
+        info "Discord notifications enabled for updates"
+    else
+        info "Discord notifications disabled (no webhook URL configured)"
+    fi
+
+    ensure_container_running
+
+    local was_running=false
+    if is_container_running; then
+        was_running=true
+        info "Container is currently running"
+    fi
+
+    while true; do
+        sleep $CHECK_INTERVAL
+
+        local is_running=false
+        if is_container_running; then
+            is_running=true
+        fi
+
+        if [ "$was_running" = true ] && [ "$is_running" = false ]; then
+            info "Container stopped - triggering restart with update check"
+            handle_container_restart
+        elif [ "$was_running" = false ] && [ "$is_running" = false ]; then
+            info "Container should be running but isn't - ensuring it's started"
+            ensure_container_running
+        fi
+
+        was_running=$is_running
+    done
+}
+
+trap 'info "Monitor shutting down"; exit 0' SIGTERM SIGINT
```

Comment:
You could move this trap to the beginning of the script. Then the main function would be unnecessary.